tests/cbor_encode/test2_simple/src/main.c

```diff
@@ -155,6 +165,66 @@ void test_pet(void)
 	zassert_mem_equal(exp_output, output, sizeof(exp_output), NULL);
 }
 
+
+void test_pet_raw(void)
```

Comment:
I might be asking for a lot here, but these tests require a lot of effort to understand for a novice user like me. If they suddenly started failing, and I were to understand what actually went wrong I think it would be good to at least have a conceptual understanding of what is going on. Would it be beneficial to, while we are here, add either a high level description of what the test does, or actually add comments to the assertions to explain what is verified. The latter might be overkill in some cases, but the first point could give a lot of value in my opinion.

Reply:
Good point. I added some comments at the top and at the asserts. I also did the same for test_pet, and for the corresponding test for decode.